cli/context_command.go

```diff
@@ -174,7 +174,15 @@ func (c *ctxCommand) editCommand(pc *fisk.ParseContext) error {
 		return err
 	}
 
-	return c.showCommand(pc)
+	// There was an error with some data in the modified config
+	// Save the known clean version and show the error
+	err = c.showCommand(pc)
+	if err != nil {
+		ctx.Save(c.name)
+		return err
```

Comment:
Looks great, lets just expand on this error a tad saying whats going on.

Tbh this should probably edit a temporary copy rather than the actual path to be safe safe butt his is a good improvement already that I am happy to merge as is.

Reply:
Sounds good. I like the idea of editing a temp copy, and we should probably do validation on all the fields as well.

This was just a short path to avoid getting locked out of editing a context via the cli.